libs/facilities/backend/facilities/src/lib/service/faclities.service.ts

```diff
@@ -0,0 +1,173 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { Aspect, XdAssetsService } from 'common-backend-insight-hub';
+import { Asset } from 'common-backend-insight-hub';
+import { PrismaService } from 'common-backend-prisma';
+import { filter, forkJoin, from, map, mergeMap, Observable, of, switchMap, toArray } from 'rxjs';
+
+@Injectable()
+export class XdFacilitesService {
+	constructor(
+		@Inject(forwardRef(() => XdAssetsService))
+		private readonly assetService: XdAssetsService,
+
+		@Inject(forwardRef(() => PrismaService))
+		private readonly prismaService: PrismaService,
+	) {}
+
+	/**
+	 * This method gets all the pumps from the Asset Management API.
+	 *
+	 * @returns An observable of the assets.
+	 */
+	public getAllPumps(): Observable<Asset[]> {
+		return this.assetService.getAssetsData().pipe(
+			map((response) => {
+				const onlyPumps = response._embedded.assets.filter(
+					(asset) => asset.typeId === 'castidev.Pump',
+				);
+
+				return onlyPumps;
+			}),
+		);
+	}
+
+	/**
+	 * This method gets the aspects for the assets from the Asset Management API.
+	 *
+	 * @param assets An observable of the assets, for which the aspects are to be fetched.
+	 * @returns An observable of the assets with their aspects.
+	 */
+	public getAspectsForAssets(assets: Observable<Asset[]>): Observable<
+		{
+			asset: Asset;
+			aspects: Aspect[];
+		}[]
+	> {
+		return assets.pipe(
+			switchMap((assets) => {
+				if (assets.length === 0) {
+					return of([]);
+				}
+
+				return forkJoin(
+					assets.map((asset) =>
+						this.assetService.getAssetAspectsData(asset.assetId).pipe(
+							switchMap((response) =>
+								of({
+									asset: asset,
+									aspects: response._embedded.aspects,
+								}),
+							),
+						),
+					),
+				);
+			}),
+		);
+	}
+
+	/**
+	 * This method upserts a single asset and its aspects in the database.
+	 *
+	 * @param asset the asset to be upserted
+	 * @param aspects the aspects of the asset
+	 * @returns the upserted asset
+	 */
+	private upsertAsset(asset: Asset, aspects: Aspect[]) {
+		return this.prismaService.asset.upsert({
+			where: {
+				assetId: asset.assetId,
+			},
+			update: {},
+			create: {
+				assetId: asset.assetId,
+				name: asset.name,
+				typeId: asset.typeId,
+				description: asset.description,
+				timeSeriesItems: {
+					createMany: {
+						data: aspects.map((aspect) => {
+							return {
+								propertySetName: aspect.name,
+								Variables: aspect.variables.map((variable) => {
```

Comment:
Why is this in CamelCase? should be 
```suggestion
								variables: aspect.variables.map((variable) => {
```

Reply:
yes i fixed this